### PR TITLE
Add docs for external routing

### PR DESCRIPTION
**Pull request by phillebaba**
Add docs for instructions how to route ingress traffic to external endpoints.

### CODE REVIEW

File: docs/xks/developer-guide/networking.md [hunk 1]
```diff
@@ -248,6 +248,89 @@ spec:
...
 ```
 
+### External Routing
+
+There is no requirement that the destination for an Ingress resource has to be served from within the cluster. It is possible to route Ingress traffic either to endpoints outside of the
+cloud provider or to another service that is only accessible from within the private network. Using the XKS Ingress instead of a separate solution has it's benefits in these situations, 
+as DNS record creation and certificate management is already setup to work. A typical use case may be during a migration period when XKS and another solution may exist in parallel. 
+All traffic can enter through XKS but then be forwarded to the external destination. The service endpoints can be updated as applications are migrated to run inside XKS instead of outside.
+
+A Service resource is required to configure the destination of the traffic. There are two options available in Kubernetes when directing traffic outside of the cluster. The first option is top
```
Comment (simongottschlag) [inline docs/xks/developer-guide/networking.md:258]: s/top/to

File: docs/xks/developer-guide/networking.md [hunk 2]
```diff
@@ -248,6 +248,89 @@ spec:
...
+kind: Service
+metadata:
+  name: to-external
+spec:
+  type: ExternalName
+  externalName: example.com
+```
+
+The other option is to create a Service which routes to a static IP. This is implemented with a Serivce without an external name or label selector, then also creating an Endpoint for the Service.
+This way the Service will only resolve to the static IP given, in this case the static IP is `1.2.3.4` and the port is `80`.
```
Comment (simongottschlag) [inline docs/xks/developer-guide/networking.md:273]: s/80/443

**Review by simongottschlag — APPROVED** [review]
Good job! 👍 I think we should write it as `Service` and `Ingress` (with backticks) instead of Service and Ingress (goes for the other resources in the written text as well).

**Review by NissesSenap — APPROVED** [review]
LGTM
How does a service work in a namespace with default deny?
Networkpolicies talks to pods not services. Is a service always able to send data to the internet?

**Comment by phillebaba** [comment]
Should not be an issue but will loop back and add some notes about network policy.

**Pull request merged**